tests/failpoints/cases/test_pending_peers.rs

```diff
@@ -151,31 +151,37 @@ fn test_on_check_busy_on_apply_peers() {
     must_get_equal(&cluster.get_engine(2), b"k2", b"v2");
 
     // Restart peer 1003 and make it busy for applying pending logs.
-    fail::cfg("on_handle_apply_1003", "return").unwrap();
+    fail::cfg("on_handle_apply_1003", "pause").unwrap();
     cluster.run_node(3).unwrap();
     let after_apply_stat = cluster.apply_state(r1, 3);
     assert!(after_apply_stat.applied_index == before_apply_stat.applied_index);
 
     // Case 1: completed regions < target count.
-    fail::cfg("on_mock_store_completed_target_count", "return").unwrap();
     sleep_ms(100);
     cluster.must_send_store_heartbeat(3);
     sleep_ms(100);
     let stats = cluster.pd_client.get_store_stats(3).unwrap();
     assert!(stats.is_busy);
-    fail::remove("on_mock_store_completed_target_count");
     sleep_ms(100);
 
     // Case 2: completed_apply_peers_count > completed_target_count but
-    //        there exists no busy peers.
+    //        there exists busy peers.
+    fail::cfg("on_mock_store_completed_target_count", "return").unwrap();
     cluster.must_send_store_heartbeat(3);
     sleep_ms(100);
     let stats = cluster.pd_client.get_store_stats(3).unwrap();
-    assert!(!stats.is_busy);
+    assert!(stats.is_busy);
+    fail::remove("on_mock_store_completed_target_count");
 
     // After peer 1003 is recovered, store also should not be marked with busy.
     fail::remove("on_handle_apply_1003");
     sleep_ms(100);
+    must_get_equal(&cluster.get_engine(3), b"k2", b"v2");
+    sleep_ms(100);
+    let after_apply_stat = cluster.apply_state(r1, 3);
+    assert!(after_apply_stat.applied_index > before_apply_stat.applied_index);
+    cluster.must_send_store_heartbeat(3);
```

Comment:
It's better to test the code path of read index as well.  Now in this test, we only rely on MsgAppend.